app/proxy/client_test.go

```diff
@@ -6,137 +6,103 @@ import (
 	"time"
 
 	"github.com/lbryio/lbrytv/app/sdkrouter"
-	"github.com/lbryio/lbrytv/config"
-	"github.com/lbryio/lbrytv/internal/lbrynet"
-	"github.com/lbryio/lbrytv/util/wallet"
+	"github.com/lbryio/lbrytv/app/wallet"
+	"github.com/lbryio/lbrytv/internal/test"
 
 	"github.com/stretchr/testify/require"
 	"github.com/ybbus/jsonrpc"
 )
 
-type MockRPCClient struct {
-	Delay        time.Duration
-	LastRequest  jsonrpc.RPCRequest
-	NextResponse chan *jsonrpc.RPCResponse
-}
-
-func NewMockRPCClient() *MockRPCClient {
-	return &MockRPCClient{
-		NextResponse: make(chan *jsonrpc.RPCResponse, 100),
-	}
-}
-
-func (c MockRPCClient) AddNextResponse(r *jsonrpc.RPCResponse) {
-	c.NextResponse <- r
-}
-
-func (c MockRPCClient) Call(method string, params ...interface{}) (*jsonrpc.RPCResponse, error) {
-	return <-c.NextResponse, nil
-}
-
-func (c *MockRPCClient) CallRaw(request *jsonrpc.RPCRequest) (*jsonrpc.RPCResponse, error) {
-	c.LastRequest = *request
-	return <-c.NextResponse, nil
-}
-
-func (c MockRPCClient) CallFor(out interface{}, method string, params ...interface{}) error {
-	return nil
-}
-
-func (c MockRPCClient) CallBatch(requests jsonrpc.RPCRequests) (jsonrpc.RPCResponses, error) {
-	return nil, nil
-}
-
-func (c MockRPCClient) CallBatchRaw(requests jsonrpc.RPCRequests) (jsonrpc.RPCResponses, error) {
-	return nil, nil
-}
-
 func TestClientCallDoesReloadWallet(t *testing.T) {
-	var (
-		r *jsonrpc.RPCResponse
-	)
-
 	rand.Seed(time.Now().UnixNano())
 	dummyUserID := rand.Intn(100)
-	rt := sdkrouter.New(config.GetLbrynetServers())
+	addr := test.RandServerAddress(t)
 
-	_, wid, _ := lbrynet.InitializeWallet(rt, dummyUserID)
-	_, err := lbrynet.WalletRemove(rt, dummyUserID)
+	err := wallet.Create(addr, dummyUserID)
+	require.NoError(t, err)
+	err = wallet.UnloadWallet(addr, dummyUserID)
 	require.NoError(t, err)
 
-	c := NewClient(rt.GetServer(wid).Address, wid, time.Second*1)
-
-	q, _ := NewQuery(newRawRequest(t, "wallet_balance", nil))
-	q.SetWalletID(wid)
-	r, err = c.Call(q)
+	q, err := NewQuery(jsonrpc.NewRequest("wallet_balance"))
+	require.NoError(t, err)
+	q.WalletID = sdkrouter.WalletID(dummyUserID)
 
+	c := NewCaller(addr, dummyUserID)
+	r, err := c.callQueryWithRetry(q)
 	// err = json.Unmarshal(result, response)
 	require.NoError(t, err)
 	require.Nil(t, r.Error)
+
+	// TODO: check that wallet is actually reloaded? what is this test even testing?
```

Comment:
It is testing a situation where a wallet is existing but not currently loaded. Checking that SDK does not respond with an error, as it would if not for the logic in `callQueryWithRetry`, is enough.